transport/src/main/java/io/netty/channel/nio/ParkingSelectStrategy.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.nio;
+
+import io.netty.channel.SelectStrategy;
+import io.netty.util.IntSupplier;
+import java.util.concurrent.locks.LockSupport;
+
+/**
+ * Custom {@link SelectStrategy} which uses {@link LockSupport#parkNanos(long)} to delay
+ * selector blocking while keeping the event loop responsive.
+ *
+ * This strategy works very well if one needs to write from outside into the event loop to avoid
+ * selector wakeup overhead.
+ */
+final class ParkingSelectStrategy implements SelectStrategy {
```

Comment:
> If that one looks good I'll get rid of the parking impl and polish it up so that we can get it ready for final review/merging?

@daschl - Yip the API looks good to me. I think we can remove this specific implementation out of the PR as you suggested.